### ISSUE TITLE
python.kak interfering colorschemes

### ISSUE THREAD

**Issue by jgmtzmd** [mawww/kakoune#1692]
yesterday on the irc channel the community helped me to set my rc file to change colorschemes acorrding to the insert/normal mode of kak.
something like this..
```
    hook global InsertBegin .* %{ colorscheme base16 } 
    hook global NormalBegin .* %{ colorscheme default }
```
it worked but i noticed that it does not when working on a python file. i disabled (by renaming) the python.kak in /usr/share/kak.. and it works again.
i wanted but could not find what in python.kak interferes with the behavior i want. i dont see there any reference to colorschemes or such things..
any help on how could i modify the python.kak to have all or most of its advantages but not to interfere with my flexible colorscheme setup?
thanks in advance.

ps: i mostly work with python. i dont know if this happens with other filetypes.

**Comment by jgmtzmd** [comment]
by the way.. dont you think that this different mode  - different colorscheme could be a distinctive and friendly default behavior of kak?
it annoys me to go constantly to the bottom-right corner of the screen to know which mode i am in. worst it is when typing some lines only to discover i was in normal mode. i think i am not the nly one who goes through this sometimes.

**Comment by Delapouite** [comment]
Possibly related : https://github.com/mawww/kakoune/issues/1171

I ended it up using danr snippet :
```
hook global InsertBegin .* %{
    echo  '-- INSERT --'
}
hook global InsertEnd .* echo
```

**Comment by jgmtzmd** [comment]
thanks delapouite. thats something i could use.
i still think different colorschemes would fit better the kakoune style of making more visible or explicit the operations. i'll just settle that as a humble recommendation to the main developers of kak.

**Comment by danr** [comment]
One caveat is that `evaluate-commands` and `execute-keys` inside `hook`s will also trigger
`InsertBegin` and `NormalBegin` (unless they have the flag `-no-hooks`).
You can start investigating if these hooks fire more often than you expect with some debug messages: 

    hook global InsertBegin .* %{ echo -debug InsertBegin }
    hook global NormalBegin .* %{ echo -debug NormalBegin }

**Comment by jgmtzmd** [comment]
@danr:
i think what you say about "evaluate-commands" ands "execute-keys" is true. if so, anytime someone wants to use InsertBegin or normalBegin in his kakrc is going to face the same problem im having now. i think we can consider this a bug, because i guess thats not the behavior we were to expect.
i ended up mapping the colorscheme to certain keystrokes..
```
hook global InsertBegin .* %{ echo -debug InsertBegin }
hook global NormalBegin .* %{ echo -debug NormalBegin }
```

but this feels really hacky. i think InsertBegin and NormalBegin hooks should only be triggered by the user. or at least we should have another hook to differentiate one from another.

**Comment by mawww** [comment]
I agree this should be fixed, what I am wondering is if `-no-hooks` should be the default for exec/eval.
